src/steamship/agents/examples/my_assistant.py

```diff
@@ -59,11 +62,13 @@ def sync_emit(blocks: List[Block], meta: Metadata):
 
         context.emit_funcs.append(sync_emit)
         self.run_agent(self._agent, context)
+        # TODO: is this right?
+        context.chat_history.append_assistant_message(output)
```

Comment:
This feels like the right direction to me. High level reactions to see if you're thinking the same way:

- The fact that `/prompt` is in user code, rather than a base class, is sort of an acknowledgement that there's some optionality here we want to keep the user in control of...
- Along with that is: what agent gets run? in what context? how are messages returned? do we remember that we said it?

It feels to me like appending to the chat history is something that we might decide is "pushed into the agent" later.. but for now it doesn't strike me that there's a downside of putting it here. 

It sort of educates the end-developer about what's actually happening, too, which can be good